src/main/scala/org/broadinstitute/dsde/firecloud/FireCloudServiceActor.scala

```diff
@@ -51,62 +55,50 @@ class FireCloudServiceActor extends HttpServiceActor {
       optionalHeaderValueByName("X-Forwarded-Host") { forwardedHost =>
         pathPrefix("") {
           pathEnd {
-            uri { uri =>
-              redirectToSwagger(forwardedHost, uri.withPath(uri.path + "api/swagger/"))
-            }
-          }
-        } ~
-          pathPrefix("swagger") {
-            pathEnd {
-              uri { uri =>
-                redirectToSwagger(forwardedHost, uri.withPath(Path("/api") ++ uri.path + "/")) }
+            cookie("access_token") { tokenCookie =>
+              serveIndex(tokenCookie.content)
             } ~
-              pathSingleSlash {
-                uri { uri =>
-                  redirectToSwagger(forwardedHost, uri.withPath(Path("/api") ++ uri.path))
-                }
-              } ~
-              serveYaml("swagger") ~ getFromResourceDirectory(swaggerUiPath)
-          }
+              complete {
+                HttpEntity(ContentType(MediaTypes.`text/html`),
+                  getResourceFileContents("swagger/auth-page.html")
+                .replace("{{googleClientId}}", FireCloudConfig.Auth.googleClientId)
+                )
+              }
+          } ~
+            pathSuffix("api-docs") {
+              complete {
+                getResourceFileContents("swagger/api-docs.yaml")
+              }
+            } ~
+            getFromResourceDirectory(swaggerUiPath)
+        }
       }
     }
   }
 
-  private def redirectToSwagger(forwardedHost: Option[String], baseUri: Uri): Route = {
-    val uri = forwardedHost match {
-      case Some(x) => baseUri.withAuthority(hostAndPortToAuthority(x))
-      case None => baseUri
+  private def serveIndex(accessToken: String): Route = {
+    val authLine = "$(function() { window.swaggerUi.api.clientAuthorizations.add(" +
+      "'key', new SwaggerClient.ApiKeyAuthorization('Authorization', 'Bearer " + accessToken +
+      "', 'header')); });"
+    val indexHtml = getResourceFileContents(swaggerUiPath + "/index.html")
+    complete {
+      HttpEntity(ContentType(MediaTypes.`text/html`),
```

Comment:
sooooo ... ... this is what we have to do to modify the default swagger-ui without copying their code, huh? Ouch.